yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -327,15 +327,74 @@ func createMigrationAssessmentStartedEvent() *cp.MigrationAssessmentStartedEvent
 func createMigrationAssessmentCompletedEvent() *cp.MigrationAssessmentCompletedEvent {
 	ev := &cp.MigrationAssessmentCompletedEvent{}
 	initBaseSourceEvent(&ev.BaseEvent, "ASSESS MIGRATION")
-	report, err := json.Marshal(assessmentReport)
+
+	sizeDetails, err := calculateSizeDetails()
+	if err != nil {
+		utils.PrintAndLog("Failed to calculate the size details of the tableIndexStats: %v", err)
+	}
+
+	finalReport := AssessMigrationPayload{
+		AssessmentJsonReport: assessmentReport,
+		SourceSizeDetails: SourceDBSizeDetails{
+			TotalIndexSize:     sizeDetails.TotalIndexSize,
+			TotalTableSize:     sizeDetails.TotalTableSize,
+			TotalTableRowCount: sizeDetails.TotalTableRowCount,
+			TotalDBSize:        source.DBSize,
+		},
+		TargetRecommendations: TargetSizingRecommendations{
+			TotalColocatedSize: sizeDetails.TotalColocatedSize,
+			TotalShardedSize:   sizeDetails.TotalShardedSize,
+		},
+		MigrationComplexity: getMigrationComplexity(), //TODO: to figure out proper algorithm
+		ConversionIssues:    schemaAnalysisReport.Issues,
+	}
+
+	finalReportBytes, err := json.Marshal(finalReport)
 	if err != nil {
-		utils.PrintAndLog("Failed to serialise the assessment report to json (ERR IGNORED): %s", err)
+		utils.PrintAndLog("Failed to serialise the final report to json (ERR IGNORED): %s", err)
 	}
 
-	ev.Report = string(report)
+	ev.Report = string(finalReportBytes)
 	return ev
 }
 
+func getMigrationComplexity() string {
+	return "NOT AVAILABLE"
+}
+
+type SizeDetails struct {
+	TotalIndexSize     int64
+	TotalTableSize     int64
+	TotalTableRowCount int64
+	TotalColocatedSize int64
+	TotalShardedSize   int64
+}
+
+func calculateSizeDetails() (SizeDetails, error) {
+	var details SizeDetails
+	colocatedTables, err := assessmentReport.GetColocatedTablesRecommendation()
+	if err != nil {
+		return details, fmt.Errorf("failed to get the colocated tables recommendation: %v", err)
+	}
+
+	if assessmentReport.TableIndexStats != nil {
+		for _, stat := range *assessmentReport.TableIndexStats {
+			if stat.IsIndex {
+				details.TotalIndexSize += safeDereferenceInt64(stat.SizeInBytes)
+			} else {
+				details.TotalTableSize += safeDereferenceInt64(stat.SizeInBytes)
+				details.TotalTableRowCount += safeDereferenceInt64(stat.RowCount)
+				if slices.Contains(colocatedTables, fmt.Sprintf("%s.%s", stat.SchemaName, stat.ObjectName)) {
+					details.TotalColocatedSize += safeDereferenceInt64(stat.SizeInBytes)
+				} else {
+					details.TotalShardedSize += safeDereferenceInt64(stat.SizeInBytes)
+				}
+			}
+		}
+	}
```

Comment:
how about defining methods on `AssessmentReport` struct for `GetTotalRowCount()`, `GetTotalColocationSize()` etc...

Reply:
I see Aneesh also had same comment, i get your point of avoid loops, it just that methods make it more structures and abstract.
Also the time complexity wise, theoretically i don't think a lot of difference - O(N) vs O(4N)

Reply:
I already made a change to separate it out in a way that will calculate size details and return all of them, so anything can be as per requirement. Also, we can more parameters to it later if required without having a separate func of each of them